pkg/kube/multicluster/secretcontroller.go

```diff
@@ -142,7 +200,14 @@ func NewController(kubeclientset kube.Client, namespace string, clusterID cluste
 	return controller
 }
 
-func (c *Controller) AddHandler(h ClusterHandler) {
+func NewHandler[T Closer](f func(cluster *Cluster, stop <-chan struct{}) T) *MultiCluster[T] {
```

Comment:
May be we can combine NewHandler and RegisterHander?

Reply:
I would but you cannot have generic methods in go unfortunately...

Or we could make it not use generics and use any instead

Reply:
I see. Trying to see if we can avoid two calls

Reply:
it can be one if it takes Controller as an argument. probably that's better

Reply:
Yes. That sounds better